kubectl.ts

```diff
@@ -41,39 +42,42 @@ export class KubectlCommand {
         this.execCommand();
     }
     async init() {
-        if (!this.kubectlbinary) {
+        if (this.kubectlbinary === tl.cwd()) {
             this.kubectlbinary = await this.downloadKubectl(this.downloadVersion);
         }
 
         this.kubectl = tl.tool(this.kubectlbinary);
     }
 
     async downloadKubectl(downloadVersion: string) : Q.Promise<string> {
-        
-        if (!downloadVersion)
-        {
-            tl.debug("getting kubectl current stable version");
-            let curl: ToolRunner = tl.tool("curl");
-            curl.arg("-LO").arg("https://storage.googleapis.com/kubernetes-release/release/stable.txt");
-            var result = curl.execSync();
-            downloadVersion = result.stdout.toString().trim();
-        }
+        try {                     
+            if (!downloadVersion) {
+                tl.debug("getting kubectl current stable version");
+                let curl: ToolRunner = tl.tool("curl");
+                curl.arg("-L").arg("https://storage.googleapis.com/kubernetes-release/release/stable.txt");
+                var result = curl.execSync();
```

Comment:
Also, why did you use execSysn() instead of using await.curl.exec(); Just curious